scanner/src/main/kotlin/scanners/fossid/FossId.kt

```diff
@@ -786,4 +868,33 @@ class FossId internal constructor(
             mapOf(SCAN_CODE_KEY to scanCode, SCAN_ID_KEY to scanId, SERVER_URL_KEY to config.serverUrl)
         )
     }
+
+    /**
+     * Return the [PurlType] as determined from the given [url], or null if there is no match. An issue will be added to
+     * [issues] in this case.
+     */
+    private fun urlToPackageType(url: String, issues: MutableList<Issue>): PurlType? = PackageProvider.get(url)?.let {
+        when (it) {
+            PackageProvider.COCOAPODS -> PurlType.COCOAPODS
+            PackageProvider.CRATES_IO -> PurlType.CARGO
+            PackageProvider.DEBIAN -> PurlType.DEBIAN
+            PackageProvider.GITHUB -> PurlType.GITHUB
+            PackageProvider.GITLAB -> PurlType.GITLAB
+            PackageProvider.GOLANG -> PurlType.GOLANG
+            PackageProvider.MAVEN_CENTRAL, PackageProvider.MAVEN_GOOGLE -> PurlType.MAVEN
+            PackageProvider.NPM_JS -> PurlType.NPM
+            PackageProvider.NUGET -> PurlType.NUGET
+            PackageProvider.PACKAGIST -> PurlType.COMPOSER
+            PackageProvider.PYPI -> PurlType.PYPI
+            PackageProvider.RUBYGEMS -> PurlType.GEM
+
+            else -> {
+                issues += FossId.createAndLogIssue(
+                    source = "FossId",
+                    message = "Cannot determine PURL type for url '$url' and provider '$it'."
```

Comment:
Instead use:

                    message = "Cannot determine PURL type for url '$url' and provider '$provider'."